src/gen9_avc_encoder.c

```diff
@@ -3596,6 +3662,456 @@ gen9_avc_set_curbe_mbenc(VADriverContextP ctx,
+    case 0:
+        /*  not use predefined search window, there should be a search_path input */
+        if ((fei_param->search_path != 0) &&
+            (fei_param->search_path != 1) &&
+            (fei_param->search_path != 2)) {
+            WARN_ONCE("Invalid input search_path for SearchWindow=0  \n");
+            assert(0);
+        }
+        /* 4 means full search, 6 means diamand search */
+        me_method = (fei_param->search_path == 1) ? 6 : 4;
+        if (((ref_width * ref_height) > 2048) || (ref_width > 64) || (ref_height > 64)) {
+            WARN_ONCE("Invalid input ref_width/ref_height in"
+                      "SearchWindow=0 case! \n");
+            assert(0);
+        }
+        break;
+
+    case 1:
+        /* Tiny - 4 SUs 24x24 window */
+        ref_width  = 24;
+        ref_height = 24;
+        len_sp     = 4;
+        break;
+
+    case 2:
+        /* Small - 9 SUs 28x28 window */
+        ref_width  = 28;
+        ref_height = 28;
+        len_sp     = 9;
+        break;
+    case 3:
+        /* Diamond - 16 SUs 48x40 window */
+        ref_width  = 48;
+        ref_height = 40;
+        len_sp     = 16;
+        break;
+    case 4:
+        /* Large Diamond - 32 SUs 48x40 window */
+        ref_width  = 48;
+        ref_height = 40;
+        len_sp     = 32;
+        break;
+    case 5:
+        /* Exhaustive - 48 SUs 48x40 window */
+        ref_width  = 48;
+        ref_height = 40;
+        len_sp     = 48;
+        break;
+    case 6:
+        /* Diamond - 16 SUs 64x32 window */
+        ref_width  = 64;
+        ref_height = 32;
+        len_sp     = 16;
+        break;
+    case 7:
+        /* Large Diamond - 32 SUs 64x32 window */
+        ref_width  = 64;
+        ref_height = 32;
+        len_sp     = 32;
+        break;
+    case 8:
+        /* Exhaustive - 48 SUs 64x32 window */
+        ref_width  = 64;
+        ref_height = 32;
+        len_sp     = 48;
+        break;
+
+    default:
+        assert(0);
+    }
+
+    /* ref_width*ref_height = Max 64x32 one direction, Max 32x32 two directions */
+    if (is_bframe) {
+        if (ref_width > 32) {
+            ref_width  = 32;
+        }
+        if (ref_height > 32) {
+            ref_height = 32;
+        }
+    }
+
+    cmd->dw0.adaptive_enable =
+        cmd->dw37.adaptive_enable = fei_param->adaptive_search;
+    cmd->dw0.t8x8_flag_for_inter_enable = cmd->dw37.t8x8_flag_for_inter_enable
+                                          = avc_state->transform_8x8_mode_enable;
+    cmd->dw2.max_len_sp = len_sp;
+    cmd->dw38.max_len_sp = 0; // HLD mandates this field to be Zero
+    cmd->dw2.max_num_su = cmd->dw38.max_num_su = 57;
+    cmd->dw3.src_access =
+        cmd->dw3.ref_access = 0; // change it to (is_frame ? 0: 1) when interlace is suppoted
+
+    if (avc_state->ftq_enable && (generic_state->frame_type != SLICE_TYPE_I)) {
+        if (avc_state->ftq_override) {
+            cmd->dw3.ft_enable = avc_state->ftq_enable;
+
```

Comment:
Not really, is there any general rule like that? I mean "no space before <else> block or something similar". There are other places in driver where we put space to improve the readability, especially for long lines. eg: there is one in i965_GetConfigAttributes().

But I an remove those if there is a generl pattern like that :)